### ISSUE TITLE
Issue with method signature userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:

### ISSUE THREAD

**Issue by kunal30xi** [Appboy/appboy-ios-sdk#113]
* [X] I have checked [Appboy's documentation](https://www.appboy.com/documentation/iOS/) and am still experiencing the issue.

# Report

## Describe your environment.

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | SDK 10.3, deployment target: 9.0
 SDK Version             | 2.29.1
 Integration Method      | carthage
 Xcode Version           | Xcode 8.3.3
 Repro rate              | all the time
 
## What did you do?
We are using carthage to pull in the AppBoy iOS SDK. As part of the unit testing effort a mock subclass of the AppBoy was created with overridden stubs of the methods we were implementing in Swift for testing.
Since our app has a deployment target of iOS 9, I'm running into an issue overriding the following method signature:

```    
class MockPushProvider: AppBoy {
    @available(iOS 10.0, *)
    override func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: (() -> Void)? = nil) {
    }
}
```

## What did you expect to happen?
Expected code to compile without issue.

## What happened instead?
Compile error reported:  
Overriding 'userNotificationCenter' must be as available as declaration it overrides

The issue is that all classes related to UserNotification are only available in iOS 10 and above. 

## Steps to reproduce

Create subclass:

```    
class MockPushProvider: AppBoy {
    @available(iOS 10.0, *)
    override func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: (() -> Void)? = nil) {
    }
}
```

A compile error is reported by Xcode.
Overriding 'userNotificationCenter' must be as available as declaration it overrides

If you were to change the method signature of the ```userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:``` method to add the  NS_AVAILABLE_IOS(10_0) macro, it would eliminate my issue and allow me to subclass the object in Swift targeting an iOS 9+ deployment.

Any help that y'all could give us would be greatly appreciated.

Thanks!

**Comment by Wenzhi** [comment]
Hey @kunal30xi,

Thanks for reporting the issue! I added the macro to the method as you asked in the newly released SDK version 2.30.0. Please feel free to reopen this issue if you still have the same trouble in the latest version of the SDK.

Thanks,
Wenzhi

**Comment by kunal30xi** [comment]
Hey @Wenzhi Thanks for getting this turned around quickly. 

I'm pulling this update using Carthage and I still seem to be getting the header from 2.29.1 even though I've updated my Appboy-iOS-SDK dependency in the Cartfile to 2.30.0.

Carthage reports `*** Downloading Appboy-iOS-SDK.framework binary at "2.30.0"` but the binary attached to the taq is still the 2.29.1 build.

I've cleared the Carthage cache (~/Library/Caches/org.carthage.CarthageKit) and removed the Carthage folder on my end multiple times but I'm still seeing the same issue.

Can you help?
